test/unittest/memory/memory_planner_validate.cpp

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file   memory_planner_validate.h
+ * @date   11 August 2021
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  Tests for memory planning
+ */
+
+#include <bitset>
+#include <random>
+#include <vector>
+
+#include <basic_planner.h>
+#include <memory_planner_validate.h>
+
+constexpr unsigned int MEM_BYTES = 50;
+constexpr unsigned int MEM_QUANT = 100;
+constexpr unsigned int INTERVAL_SIZE = 5;
+
+void MemoryPlannerValidate::SetUp() {
+  plan_type = GetParam();
+
+  planner = nullptr;
+  if (plan_type == nntrainer::BasicPlanner::type)
+    planner = std::move(std::make_unique<nntrainer::BasicPlanner>());
+  else
+    throw std::invalid_argument("Invalid planner type");
+}
+
+/**
+ * @brief Validate the provided layout does not overlap
+ */
+static bool validateNoOverlap(const std::vector<size_t> &memory_size,
+                              const std::vector<size_t> &memory_offset,
+                              unsigned int size) {
+  std::vector<bool> mem_overlap(size, false);
+  for (unsigned int idx = 0; idx < memory_size.size(); idx++) {
+    for (unsigned int mem = memory_offset[idx];
+         mem < memory_offset[idx] + memory_size[idx]; mem++) {
+      EXPECT_FALSE(mem_overlap[mem]);
+      if (mem_overlap[mem])
+        return false;
+      mem_overlap[mem] = true;
+    }
+  }
+
+  /** This ensures that there are no unnecessary gaps in between */
+  EXPECT_EQ(mem_overlap.size(),
+            std::accumulate(mem_overlap.begin(), mem_overlap.end(), 0));
+
+  return true;
+}
+
+/**
+ * @brief Validate the provided layout does completely overlap
+ */
+static bool validateAllOverlap(const std::vector<size_t> &memory_size,
+                               const std::vector<size_t> &memory_offset) {
+  for (unsigned int idx = 0; idx < memory_size.size(); idx++) {
+    EXPECT_EQ(memory_offset[idx], memory_offset[0]);
+  }
+
+  return true;
+}
+
+/**
+ * @brief Validate the provided layout does not overflow outside the given
+ * size
+ */
+static bool validateOverflow(const std::vector<size_t> &memory_size,
+                             const std::vector<size_t> &memory_offset,
+                             size_t pool_size) {
+  for (unsigned int idx = 0; idx < memory_size.size(); idx++)
+    if (memory_offset[idx] + memory_size[idx] > pool_size)
+      return false;
+
+  return true;
+}
+
+/**
+ * @brief Test memory planning with full overlap of requested memories
+ */
+TEST_P(MemoryPlannerValidate, full_overlap) {
```

Comment:
could you clarify by renaming validityIsFullyOverlapping?

Reply:
added as details in doxygen (kept name same for consistency with other unittests).

Reply:
Cool